lib/semantics/check-do.cc

```diff
@@ -434,6 +455,40 @@ class DoContext {
 public:
   DoContext(SemanticsContext &context) : context_{context} {}
 
+  // Mark this DO construct as a point of definition for the DO variables
+  // or index-names it contains.  If they're already defined, emit an error
+  // message.  We need to remember both the variable and the source location of
+  // the variable in the DO construct so that we can remove it when we leave
+  // the DO construct and use its location in error messages.
+  void DefineDoVariables(const parser::DoConstruct &doConstruct) {
+    if (doConstruct.IsDoNormal()) {
+      context_.ActivateDoVariable(GetDoVariable(doConstruct));
+    } else if (doConstruct.IsDoConcurrent()) {
+      auto &loopControl{doConstruct.GetLoopControl()};
+      if (loopControl) {
```

Comment:
You could write these two lines as:
`if (const auto &loopControl{doConstruct.GetLoopControl()}) {`

You should use `const auto &` rather than just `auto &`.

Reply:
Thanks, Tim.  I'll change that.